test/unit_tests/container/container.cpp

```diff
@@ -31,11 +67,183 @@ struct handle_interface: handle_base<Identifier> {
+    data.set_field_status<field::DOff>(false);
+
+    auto c = field::C{};
+
+    for (size_t field_index = 0; field_index < c.num_variables(); ++field_index) {
+        CHECK(data.template get_array_dims<field::C>()[field_index] ==
+              c.array_dimension(field_index));
+        CHECK(data.template get_array_dims<field::C>(field_index) ==
+              c.array_dimension(field_index));
+    }
+
+    CHECK(data.template get_array_dims<field::DOff>(0) == 1ul);
+    CHECK(data.template get_array_dims<field::DOn>(0) == 1ul);
+}
+
+TEST_CASE("soa::get_num_variables", "[Neuron][data_structures]") {
+    storage data;
+
+    data.set_field_status<field::DOn>(true);
+    data.set_field_status<field::DOff>(false);
+
+    auto c = field::C{};
+
+    CHECK(data.get_num_variables<field::C>() == c.num_variables());
+    CHECK(data.get_num_variables<field::DOff>() == 1ul);
+    CHECK(data.get_num_variables<field::DOn>() == 1ul);
+}
+
+TEST_CASE("Identifier defer delete ", "[Neuron][internal][data_structures]") {
+    storage data;
+
+    REQUIRE(detail::identifier_defer_delete_storage != nullptr);
+    auto usage_before = detail::compute_identifier_defer_delete_storage_size();
+    { owning_handle instance{data}; }
+    auto usage_after = detail::compute_identifier_defer_delete_storage_size();
+
+    CHECK(usage_after.size - usage_before.size > 0);
+    CHECK(usage_after.capacity > 0);
+    CHECK(usage_before.size <= usage_before.capacity);
+    CHECK(usage_after.size <= usage_after.capacity);
+}
+
+TEST_CASE("defer delete storage pointer", "[Neuron][internal][data_structures]") {
+    REQUIRE(detail::defer_delete_storage != nullptr);
+
+    auto usage_before = detail::compute_defer_delete_storage_size();
+    { storage data; }
+    auto usage_after = detail::compute_defer_delete_storage_size();
+
+    CHECK(usage_after.size - usage_before.size > 0);
+    CHECK(usage_after.capacity > 0);
+    CHECK(usage_before.size <= usage_before.capacity);
+    CHECK(usage_after.size <= usage_after.capacity);
+}
+
+template <class Tag, class Storage>
+std::size_t compute_row_size(const Storage& data) {
+    std::size_t local_size = 0ul;
+    auto tag = data.template get_tag<Tag>();
+    for (int field_index = 0; field_index < detail::get_num_variables(tag); ++field_index) {
+        local_size += data.template get_array_dims<Tag>()[field_index] * sizeof(typename Tag::type);
+    }
+
+    return local_size;
+}
+
+TEST_CASE("container memory usage", "[Neuron][internal][data_structures]") {
+    storage data;
+    data.set_field_status<field::DOn>(true);
+    data.set_field_status<field::DOff>(false);
+
+    std::size_t row_size = compute_row_size<field::A>(data) + compute_row_size<field::B>(data) +
+                           compute_row_size<field::C>(data) + compute_row_size<field::DOn>(data);
+
+    auto r1 = owning_handle{data};
+    auto r2 = owning_handle{data};
+    auto r3 = owning_handle{data};
+
+    auto n_rows = data.size();
```

Comment:
Could check/require this is `== 3` too if you want (right?)

Reply:
I don't think I have any reason to distrust `data.size()` in this test. We can test that elsewhere.